opentracing/tracecontext.go

```diff
@@ -51,18 +43,18 @@ type TraceContext interface {
 	// `restrictedKey` MUST match the regular expression
 	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
 	// start with a letter or number, and the remaining characters must be
-	// letters, numbers, or hyphens. See CanonicalizeTraceTagKey(). If
-	// `restrictedKey` does not meet these criteria, SetTraceTag() results in
-	// undefined behavior.
+	// letters, numbers, or hyphens. See CanonicalizeTraceAttributeKey(). If
+	// `restrictedKey` does not meet these criteria, SetTraceAttribute()
+	// results in undefined behavior.
 	//
 	// Returns a reference to this TraceContext for chaining, etc.
-	SetTraceTag(restrictedKey, value string) TraceContext
+	SetTraceAttribute(restrictedKey, value string) TraceContext
```

Comment:
So, imagine this tree of spans:

```
              A
              |
         +----+----+
         |         |
         B         C
         |         |
      +--+--+   +--+--+
      |     |   |     |
      D     E   F     G
```

If someone decides to set an attribute "upstream" in `F`, `D` or `E` may have already finished... And of course there's no formal requirement (in the data model) that children finish before their parents. So I do think the upstream case is more complex than the downstream case. If we want to try to future-proof the API, we could call it `PropagateAttributeDown` or `PropagateAttributeDownstream`?? A mouthful for sure, though I'm comforted by how infrequently this will be called.